Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorHelpers.cpp

```diff
@@ -173,6 +185,12 @@ namespace AzToolsFramework
             }
         }
 
+        // Verify if the entity Id corresponds to an entity that is focused; if not, halt selection.
+        if (!m_focusModeInterface->IsInFocusSubTree(entityIdUnderCursor))
```

Comment:
question: Have we done any profiling of this call? What is the overhead of calling this for every entity in the view every frame? I'm just worried in large scenes this might start to be a problem

Reply:
Sure understood, but would be good just to check it's usable with a large number (this can be adhoc, just in the editor, but it's always good to sanity check it in case). Thanks!

Reply:
Since this is more of a concern with the whole system over just this PR, I am splitting up that investigation to its own task so that it doesn't stop this from being merged. I will work on setting up Pix and other automated tests to provide data on the performance hit that can drive optimization efforts.